### DISCUSSION TITLE
definePageMeta cannot access const

### DISCUSSION THREAD

**Discussion by tasiotas** [nuxt/nuxt#17788] [Questions] [answered]
Hi,

Why I cannot access slug variable defined above definePageMeta?
I have a feeling its because its a macro(just repeating what I read in other posts), but to be honest, I don't know what it means or what limitations it imposes.

As you can see, Im trying to set page title based on route slug.

Thanks

```
<script setup>
  const route = useRoute()
  const slug = route.params.slug

  definePageMeta({
    title: slug,
  })
</script>
```

**Comment by danielroe** [comment, marked as answer]
The issue here is that `useRoute()` can only be run in the setup function of a composable, but definePageMeta is extracted from this context (it's not being run within the composable). This is similar to `defineProps` and `defineEmits` from `vue` itself.

In your case I think you actually want `useMeta`:

```vue
<script setup>
  const route = useRoute()
  const slug = route.params.slug

  useMeta({
    title: slug,
  })
</script>
```

**Reply by tasiotas** [reply]
Thanks for clarifications. Here is grand picture of what I am after.
I was hoping to extend this example of defining page title. 
https://v3.nuxtjs.org/docs/usage/meta-tags/#example-usage-with-definepagemeta

Wanted to cover a case where page doesnt have title, so it falls back to default one.
Is that a valid approach in Nuxt?

pages/some-page.vue
```
<script setup>
definePageMeta({
  title: 'Some Page'
})
</script>
```

layouts/default.vue
```
<script setup>
  const route = useRoute()
  let title = route.meta.title
  if (title) {
    title = title + " - Company Name"
  } else {
    title = "Company Name"
  }
useMeta({
  title: title)
})
</script>
```

**Reply by danielroe** [reply]
Yes, looks good. Although you'd probably make title a computed property in your layout...

```vue
<script setup>
  const route = useRoute()
  const title = computed(() => route.meta.title ? title + " - Company Name" : "Company Name")
  useMeta({
    title
  })
</script>
```

**Reply by tasiotas** [reply]
thank you
